### PR TITLE
extract HttpTestSuite from skinny-http

### PR DESCRIPTION
**Pull request by Kuchitama**
The test code of skinny-http is very good as a test of http client.
I'd like to use the test code in other projects.

So, I extracted the test code as skinny-http-test-suite.

### CODE REVIEW

**Comment by seratch** [comment]
The concept looks fine but these code should be more sophisticated. Since I wrote them just for testing skinny-http-client library, the API design and details are not mature enough.

File: http-test-suite/src/main/scala/skinny/http/test/server/PostFormdataServer.scala [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+              val upload = new ServletFileUpload(factory)
+              upload.setSizeMax(20 * 1024)
+              upload.setFileSizeMax(10 * 1024)
+
+              var toResponse = "FAILURE!!!"
+              try {
+                  import scala.collection.JavaConverters._
+                  val items: java.util.List[FileItem] = upload.parseRequest(req)
+                  for (item <- items.asScala) {
+                      if ("toResponse".equals(item.getFieldName())) {
```
Comment (seratch) [inline http-test-suite/src/main/scala/skinny/http/test/server/PostFormdataServer.scala]: This field name is not useful for general purposes. We should create some customizable points instead.

**Comment by seratch** [comment]
These code basically depend on only Java things. Just an idea but writing this completely in Java might be better in the aspect of portability among Scala versions in the future.

File: http-test-suite/src/main/scala/skinny/http/test/server/PostFormdataServer.scala (hunk 1 above)
Comment (Kuchitama) [inline http-test-suite/src/main/scala/skinny/http/test/server/PostFormdataServer.scala]: Oops. 
We need change here.

**Comment by Kuchitama** [comment]
Thank you for your review.

> the API design and details are not mature enough.

I think so too.
But this is enough useful to test any http clients.

I think that we can provide this feature with a few changes.

**Comment by seratch** [comment]
@Kuchitama I think skinny-micro-server looks useful enough to meet your demand. https://github.com/skinny-framework/skinny-micro/blob/master/micro-server/src/test/scala/example/WebAppSpec.scala

I'd like to close this PR without merging it if you don't mind. What do you think?

**Comment by Kuchitama** [comment]
I see. 
skinny-micro-server is very useful.

**Pull request closed without merging**